### PR TITLE
Add vector store integration of lindorm, including knn search, …

### PR DESCRIPTION
**Pull request by Rainy-GG**
# Description

[Lindorm](https://www.alibabacloud.com/help/en/lindorm) is a cloud native multi-model database service developed by Alibaba Cloud. It allows you to store data of all sizes. Lindorm supports low-cost storage and processing of large amounts of data and the pay-as-you-go billing method. It is compatible with the open standards of multiple open source software, such as Apache HBase, Apache Cassandra, Apache Phoenix, OpenTSDB, Apache Solr, and SQL.
Here we add vector store integration with Lindorm.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by Rainy-GG** [comment]
@logan-markewich 😊 Hi Logan～

I have fixed some coding standard issues and the unit testing errors that occurred in the workflow. Could you please review this PR and re-run the action trigger if necessary?

Thanks!🙏

File: llama-index-integrations/vector_stores/llama-index-vector-stores-lindorm/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+description = "llama-index vector_stores lindorm integration"
+exclude = ["**/BUILD"]
+license = "MIT"
+name = "llama-index-vector-stores-lindorm"
+readme = "README.md"
+version = "0.1.0"
+
+[tool.poetry.dependencies]
+python = ">=3.8.1,<4.0"
+llama-index-core = "^0.10.1"
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-lindorm/pyproject.toml:34]: We can add opensearch-py as a dependency here (and anything else I missed!)

Comment (Rainy-GG) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-lindorm/pyproject.toml:34]: Thanks for your reminder! 😊 I have added `opensearch-py` and ensured it follows our project's style conventions. Here is the updated configuration in `pyproject.toml`:

```toml
[tool.poetry.dependencies]
python = ">=3.8.1,<4.0"
llama-index-core = "^0.10.1"
opensearch-py = { version = "^2.4.2", extras = ["async"] }

Comment (Rainy-GG) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-lindorm/pyproject.toml:34]: @logan-markewich  Hi Logan, could you please trigger the workflows and review the changes when you have a moment? Thanks! 😊

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**